jobs/v3/snippets/general_search_sample.go

```diff
@@ -0,0 +1,203 @@
+
+// This is a quickstart sample of using the Google Cloud Job Discovery API.
+package main
+
+import (
+	"os"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	jobs "google.golang.org/api/jobs/v3"
+)
+
+const requestDeadline = 10 * time.Second
+
+func prettyFormat(v interface{}) string {
+	j, err := json.MarshalIndent(v, "", "    ")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(j)
+}
+
+func makeCompany() *jobs.Company {
+	return &jobs.Company{
+		DisplayName:          "Google",
+		HeadquartersAddress:           "1600 Amphitheatre Parkway Mountain View, CA 94043",
+		ExternalId: fmt.Sprintf("company:%d", rand.New(rand.NewSource(time.Now().UnixNano())).Int63()),
+	}
+}
+
+
+func createCompany(ctx context.Context, js *jobs.Service, toCreate *jobs.Company, parent string) (*jobs.Company, error) {
+	fmt.Println("Attempting to create a Company...")
+
+	createCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	request := &jobs.CreateCompanyRequest {
+		Company: toCreate,
+	}
+
+	c, err := js.Projects.Companies.Create(parent,request).Context(createCtx).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("Company created:\n %v\n", prettyFormat(c))
+	return c, nil
+}
+
+
+func deleteCompany(ctx context.Context, js *jobs.Service, name string) error {
+	fmt.Printf("Attempting to delete a Company with name %s...\n", name)
+
+	deleteCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	if _, err := js.Projects.Companies.Delete(name).Context(deleteCtx).Do(); err != nil {
+		return err
+	}
+
+	fmt.Println("Company deleted")
+	return nil
+}
+
+func makeJob(companyName string) *jobs.Job {
+	return &jobs.Job{
+		RequisitionId:   fmt.Sprintf("job:%d", rand.New(rand.NewSource(time.Now().UnixNano())).Int63()),
+		Title:        "Software Engineer",
+		CompanyName:     companyName,
+		ApplicationInfo: &jobs.ApplicationInfo{Uris: []string{"http://careers.google.com"}},
+		Description:     "Design, develop, test, deploy, maintain and improve software.",
+	}
+}
+
+func createJob(ctx context.Context, js *jobs.Service, toCreate *jobs.Job, parent string) (*jobs.Job, error) {
+	fmt.Println("Attempting to create a Job...")
+
+	createCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	j, err := js.Projects.Jobs.Create(parent, &jobs.CreateJobRequest{Job: toCreate}).Context(createCtx).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("Job created:\n %v\n", prettyFormat(j))
+	return j, nil
+}
+
+func deleteJob(ctx context.Context, js *jobs.Service, name string) error {
+	fmt.Printf("Attempting to delete a Job with name %s...\n", name)
+
+	deleteCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	if _, err := js.Projects.Jobs.Delete(name).Context(deleteCtx).Do(); err != nil {
+		return err
+	}
+
+	fmt.Println("Job deleted")
+	return nil
+}
+
+// [START basic_keyword_search]
```

Comment:
Do we need "basic" in here? Could this be "keyword_search" (same with the function name, etc.)? Doesn't look basic to me! :)